pkg/controller/servicing/lifecycler.go

```diff
@@ -0,0 +1,359 @@
+			node.GetName())
+	} else {
+		lc.Recorder.Eventf(
+			lc.Kluster,
+			core_v1.EventTypeNormal,
+			events.FailedDrainNode,
+			"Preparing upgrade for node: %v. Failed to drain node: %v",
+			node.GetName(),
+			err)
+	}
+	return err
+}
+
+// Reboot writes an Event
+func (lc *EventingLifeCycler) Reboot(node *core_v1.Node) error {
+	err := lc.LifeCycler.Reboot(node)
+	if err == nil {
+		lc.Recorder.Eventf(
+			lc.Kluster,
+			core_v1.EventTypeNormal,
+			events.SuccessfulRebootNode,
+			"Upgrading OS for node: %v. Reboot successful.",
+			node.GetName())
+	} else {
+		lc.Recorder.Eventf(
+			lc.Kluster,
+			core_v1.EventTypeNormal,
+			events.FailedRebootNode,
+			"Upgrading OS for node: %v. Reboot failed: %v",
+			node.GetName(),
+			err)
+	}
+	return err
+}
+
+// Replace writes an Event
+func (lc *EventingLifeCycler) Replace(node *core_v1.Node) error {
+	err := lc.LifeCycler.Reboot(node)
+	if err == nil {
+		lc.Recorder.Eventf(
+			lc.Kluster,
+			core_v1.EventTypeNormal,
+			events.SuccessfulReplaceNode,
+			"Replacing node for upgrade: %v. Termination successful",
+			node.GetName())
+	} else {
+		lc.Recorder.Eventf(
+			lc.Kluster,
+			core_v1.EventTypeNormal,
+			events.FailedReplaceNode,
+			"Replacing node for upgrade: %v. Termination failed: %v",
+			node.GetName(),
+			err)
+	}
+	return err
+}
+
+// Drain collects metrics
+func (lc *InstrumentingLifeCycler) Drain(node *core_v1.Node) (err error) {
+	defer func(begin time.Time) {
+		lc.Latency.With(
+			prometheus.Labels{
+				"controller": "servicing",
+				"method":     "Drain",
+			}).Observe(time.Since(begin).Seconds())
+
+		lc.Total.With(
+			prometheus.Labels{
+				"controller": "servicing",
+				"method":     "Drain",
+			}).Add(1)
+
+		if err != nil {
+			lc.Failed.With(
+				prometheus.Labels{
+					"controller": "servicing",
+					"method":     "Drain",
+				}).Add(1)
+		} else {
+			lc.Successful.With(
+				prometheus.Labels{
+					"controller": "servicing",
+					"method":     "Drain",
+				}).Add(1)
+		}
+	}(time.Now())
+	return lc.LifeCycler.Drain(node)
+}
+
+// Reboot collects metrics
+func (lc *InstrumentingLifeCycler) Reboot(node *core_v1.Node) (err error) {
+	defer func(begin time.Time) {
+		lc.Latency.With(
+			prometheus.Labels{
+				"controller": "servicing",
+				"method":     "Reboot",
+			}).Observe(time.Since(begin).Seconds())
+
+		lc.Total.With(
+			prometheus.Labels{
+				"controller": "servicing",
+				"method":     "Reboot",
+			}).Add(1)
+
+		if err != nil {
+			lc.Failed.With(
+				prometheus.Labels{
+					"controller": "servicing",
+					"method":     "Reboot",
+				}).Add(1)
+		} else {
+			lc.Successful.With(
+				prometheus.Labels{
+					"controller": "servicing",
+					"method":     "Reboot",
+				}).Add(1)
+		}
+	}(time.Now())
+	return lc.LifeCycler.Reboot(node)
+}
+
+// Replace collects metrics
+func (lc *InstrumentingLifeCycler) Replace(node *core_v1.Node) (err error) {
+	defer func(begin time.Time) {
```

Comment:
the `prometheus.Labels` struct could be defined once in the beginning and passed to each `With` method